pymc_marketing/clv/models/beta_geo_beta_binom.py

```diff
@@ -0,0 +1,677 @@
+            t=[2, 5, 7, 10],
+        )
+
+    References
+    ----------
+    .. [1] Peter Fader, Bruce Hardie, and Jen Shang.
+           "Customer-Base Analysis in a Discrete-Time Noncontractual Setting".
+           Marketing Science, Vol. 29, No. 6 (Nov-Dec, 2010), pp. 1086-1108.
+           https://www.brucehardie.com/papers/020/fader_et_al_mksc_10.pdf
+    """
+
+    _model_type = "BG/BB"  # Beta-Geometric, Beta-Binomial Distribution
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        *,
+        model_config: ModelConfig | None = None,
+        sampler_config: dict | None = None,
+    ):
+        super().__init__(
+            data=data,
+            model_config=model_config,
+            sampler_config=sampler_config,
+            non_distributions=None,
+        )
+        self._validate_cols(
+            data,
+            required_cols=[
+                "customer_id",
+                "frequency",
+                "recency",
+                "T",
+            ],
+            must_be_unique=["customer_id"],
+            must_be_homogenous=["T"],
+        )
+
+    @property
+    def default_model_config(self) -> ModelConfig:
+        return {
+            "alpha_prior": None,
+            "beta_prior": None,
+            "gamma_prior": None,
+            "delta_prior": None,
+            "phi_purchase_prior": Prior("Uniform", lower=0, upper=1),
+            "kappa_purchase_prior": Prior("Pareto", alpha=1, m=1),
+            "phi_dropout_prior": Prior("Uniform", lower=0, upper=1),
+            "kappa_dropout_prior": Prior("Pareto", alpha=1, m=1),
+        }
+
+    def build_model(self) -> None:  # type: ignore[override]
+        coords = {
+            "obs_var": ["recency", "frequency"],
+            "customer_id": self.data["customer_id"],
+        }
+        with pm.Model(coords=coords) as self.model:
+            # purchase rate priors
+            if (
+                self.model_config["alpha_prior"] is None
+                or self.model_config["beta_prior"] is None
+            ):
+                # hierarchical pooling of purchase rate priors
+                phi_purchase = self.model_config["phi_purchase_prior"].create_variable(
+                    "phi_purchase"
+                )
+                kappa_purchase = self.model_config[
+                    "kappa_purchase_prior"
+                ].create_variable("kappa_purchase")
+
+                alpha = pm.Deterministic("alpha", phi_purchase * kappa_purchase)
+                beta = pm.Deterministic("beta", (1.0 - phi_purchase) * kappa_purchase)
+            else:
+                alpha = self.model_config["alpha_prior"].create_variable("alpha")
+                beta = self.model_config["beta_prior"].create_variable("beta")
+
+            # dropout priors
+            if (
+                self.model_config["gamma_prior"] is None
+                or self.model_config["delta_prior"] is None
```

Comment:
Same suggestion here. Would that make it so that none of the dictionary values will have to be none

What are your thoughts?